### PR TITLE
unnominated2019 filter

### PR DESCRIPTION
**Pull request by Raemon**
This attempts to make a post filter that shows only "non-nominated posts" for 2019. 

It currently is blocked on a bug in the migration code. @jimrandomh, @Discordius had said in slack you should look at that. When I attempt to run the migration I get an error which Habryka said seemed like a bug in some underlying related code, not sure why.

Error that results from running `Vulcan.migrations.nominationCount2019()`

```
W20201207-15:54:03.465(-8)? (STDERR) FAILED migration: nominationCount2019.
W20201207-15:54:03.467(-8)? (STDERR) TypeError: Cannot read property 'extraLoaders' of undefined
W20201207-15:54:03.467(-8)? (STDERR)     at packages/lesswrong/lib/loaders.ts:22:16
W20201207-15:54:03.467(-8)? (STDERR)     at /Users/raymondarnold/.meteor/packages/promise/.0.11.2.7j140w.ob2tj++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
W20201207-15:54:03.468(-8)? (STDERR)  => awaited here:
W20201207-15:54:03.468(-8)? (STDERR)     at Function.Promise.await (/Users/raymondarnold/.meteor/packages/promise/.0.11.2.7j140w.ob2tj++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
W20201207-15:54:03.468(-8)? (STDERR)     at packages/lesswrong/lib/utils/schemaUtils.ts:382:29
W20201207-15:54:03.468(-8)? (STDERR)     at /Users/raymondarnold/.meteor/packages/promise/.0.11.2.7j140w.ob2tj++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
W20201207-15:54:04.743(-8)? (STDERR)  => awaited here:
```

### CODE REVIEW

**Comment by jimrandomh** [comment]
Caused by poor typechecking around functions on schemas. Will improve the typechecking and fix the bug in the process, but it'll be downstream of this branch: https://github.com/LessWrong2/Lesswrong2/pull/3770 and will have to wait for that to be reviewed.

**Comment by Raemon** [comment]
It's fairly urgent to get this up before the nominations period is over. Is it possible for me (or you) to get the fixes locally, and run the migration locally, faster than the overall review process might be?

**Comment by jimrandomh** [comment]
Aight, fixed the bug without worrying about the type annotations for now. The migration should work now, and I have it running right now.

**Comment by Raemon** [comment]
Doublechecking – does this mean I have to do anything else with the
migration, or have you taken care of it in it's entirety?

**Comment by jimrandomh** [comment]
With that fix applied, running the migration didn't crash, but it also didn't seem to update the database. So there's a second bug, which I'm looking into now.

(The thing that's different that's exposing the bug now, btw, is that the field is a `denormalizedCountOfReferences`, where before we've only used these migration utility functions on things with `schemaDefaultValue`.)

**Comment by jimrandomh** [comment]
So the issue is that it's not just filling in 0s, it's doing a query to find the true nomination count, and while doing that query, it uses loaders. Loaders are supposed to be ephemeral things which last only for one request, and they have caching where the eviction strategy is "discard everything at the end of the request but otherwise grow without bound". In the context of a migration script, that leads to running out of memory.

File: packages/lesswrong/components/review/FrontpageReviewPhase.tsx [hunk 1]
```diff
@@ -139,7 +138,7 @@ const FrontpageReviewPhase = ({classes}) => {
         </LWTooltip>}
       >
         <LWTooltip title="All Posts written in 2019 are eligible to participate in the review. Click here to see all posts written in 2019.">
-          <Link to={"/allPosts?timeframe=yearly&after=2019-01-01&before=2020-01-01&limit=100&sortedBy=top"}><SettingsButton showIcon={false} label="See All 2019 Posts"/></Link>
+          <Link to={"/allPosts?timeframe=yearly&after=2019-01-01&before=2020-01-01&limit=100&sortedBy=top?filter=unnominated2019"}><SettingsButton showIcon={false} label="Nominate 2019 Posts"/></Link>
```
Comment (jimrandomh) [inline packages/lesswrong/components/review/FrontpageReviewPhase.tsx:141]: This should be `&filter=` not `?filter=` (the rule is that the first URL parameter is a `?` and all subsequent ones are `&`).

Comment (jimrandomh) [reply packages/lesswrong/components/review/FrontpageReviewPhase.tsx:141]: I think you probably also want to add this as a filter option in `PostsListSettings`. I'm not sure what happens if you don't.

Comment (Raemon) [reply packages/lesswrong/components/review/FrontpageReviewPhase.tsx]: "untagged" and other weird one-of filters don't appear there. I think that's just for making it show up super legibly in the allPosts page, which I don't think we want.

**Review by jimrandomh — APPROVED** [review]

**Pull request merged**